native/mod_proxy_cluster/mod_proxy_cluster.c

```diff
@@ -1510,17 +1562,40 @@ static void update_workers_lbstatus(proxy_server_conf *conf, apr_pool_t *pool, s
                 apr_status_t rv;
                 apr_pool_t *rrp;
                 request_rec *rnew;
-                proxy_worker *worker = get_worker_from_id_stat(conf, id[i], stat);
+                proxy_worker *worker;
+                apr_thread_mutex_lock(lock);
+                worker = get_worker_from_id_stat(conf, id[i], stat, ou);
+                apr_thread_mutex_unlock(lock);
+
                 if (worker == NULL)
                     continue; /* skip it */
 #if AP_MODULE_MAGIC_AT_LEAST(20101223,1)
                 apr_snprintf(sport, sizeof(sport), ":%d", worker->s->port);
+
+                if (strcmp(worker->s->scheme, ou->mess.Type) ||
+                    strcmp(worker->s->hostname, ou->mess.Host) ||
+                    strcmp(sport, ou->mess.Port)) {
+                    /* the worker doesn't correspond to the node */
+                    worker->s->index = 0;
+                    /* XXX: broken ap_my_generation--; mark old generation that will recreate the process */ 
```

Comment:
@jfclere Hmm, how exactly this commented out section ```/* XXX: broken ap_my_generation--; mark old generation that will recreate the process */``` addresses concerns raised in https://github.com/modcluster/mod_cluster/pull/193?

The main reason for merging https://github.com/modcluster/mod_cluster/pull/189 and not going forward with https://github.com/modcluster/mod_cluster/pull/193 was the inability to port ```ap_my_generation--;``` hack to httpd 2.4.

I might be missing something. Does this commit work as expected with httpd 2.4.x?

If the conclusion drawn on https://github.com/modcluster/mod_cluster/pull/193 is wrong and it can in fact work, please share the explanation for both https://github.com/modcluster/mod_cluster/pull/193 and  https://github.com/modcluster/mod_proxy_cluster/pull/12/.